src/platform/nrfconnect/wifi/NrfWiFiDriver.cpp

```diff
@@ -52,6 +52,11 @@ CHIP_ERROR NrfWiFiDriver::Init(NetworkStatusChangeCallback * networkStatusChange
 
 void NrfWiFiDriver::OnNetworkStatusChanged(Status status)
 {
+    if (status == Status::kSuccess)
+    {
+        ConnectivityMgr().SetWiFiStationMode(ConnectivityManager::kWiFiStationMode_Enabled);
```

Comment:
I think we may map the status to `ConnectivityManager::WiFiStationMode::ENABLED/DISABLED` to post the disconnect event in case the station is disconnected. On the other hand I am not sure if in principle we can assume the fail status here always shall indicate the disconnection (i.e. when the station has not been even connected beforehand).

Reply:
Yeah I am not sure about it either. What would the use case for us to handle the disconnection event? The status LED shows provisioning state and not current connection state with AP.

Reply:
Ok, so let's leave it as it is for now.